src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -146,11 +147,28 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
    */
   ColumnID get_column_id(const AbstractExpression& expression) const;
 
+  /**
+   * Searches all output expressions.
+   * @return A shared pointer to the LQPColumnExpression with @param column_id or std::nullopt if it cannot be found.
+   */
+  std::optional<const std::shared_ptr<LQPColumnExpression>> find_column_expression(const ColumnID column_id) const;
+
   /**
    * @return whether the output column at @param column_id is nullable
    */
   virtual bool is_column_nullable(const ColumnID column_id) const;
 
+  /**
+   * @return A shared pointer to a vector of unique constraints valid for the current LQP.
+   * See lqp_unique_constraint.hpp for documentation. Nodes that query data must override this function.
+   */
+  [[nodiscard]] virtual std::shared_ptr<LQPUniqueConstraints> unique_constraints() const;
```

Comment:
No. I have put it up somewhere in the codebase and use it for any "expensive getter-functions".

Reply:
I looked into this and read https://www.bfilipek.com/2017/11/nodiscard.html

These cases make sense to me:

> * not using the return value always is a “huge mistake” (e.g. always resulting in resource leak)
> * not using the return value is a source of trouble and easily can happen (not obvious that something is wrong) 

I am not a fan of

> * not using the return value is usually an error.

as I worry that this might spiral into a long discussion on where this applies and where it doesn't. It is clear in the present case that calling the method and not using the result is an error and has a negative impact on performance. Where do we stop though? A node's hash function might be cheaper than the one discussed above, as it does not recurse through the entire LQP. What about `description()`, which has no recursion but builds a string? Adding nodiscard to all of these would be noisy and hard to enforce. Adding it only to "expensive" cases would add a significant burden on the developers to identify these with only little benefit. I can't remember doing a code review where an accidentally discarded return value was an issue.

As for the existing uses in the TPC-C benchmark, they fall into the second category mentioned above. `execute()` sounds like something that stands for itself. If, however, you do not check for the return code, you might expect a returned call to mean that the procedure was executed. That is not the case, as transaction conflicts might mean that your call to `execute()` did not have any visible side effects.

@hyrise/zweirise-developers

Reply:
> As for the existing uses in the TPC-C benchmark, they fall into the second category mentioned above. execute() sounds like something that stands for itself. If, however, you do not check for the return code, you might expect a returned call to mean that the procedure was executed. 

Good comment. I'm ok with this as a convention. 👍